packages/framework/package.json

```diff
@@ -24,12 +24,13 @@
     "lint:fix": "pnpm lint -- --fix",
     "format": "prettier --check --ignore-path .gitignore .",
     "format:fix": "prettier --write --ignore-path .gitignore .",
-    "build": "tsup",
+    "build": "tsup && pnpm check:circulars",
     "build:watch": "tsup --watch",
     "$comment:bump:prerelease": "This is a workaround to support `npm version prerelease` with lerna",
     "bump:prerelease": "npm version prerelease --preid=alpha & PID=$!; (sleep 1 && kill -9 $PID) & wait $PID",
     "release:alpha": "pnpm bump:prerelease || pnpm build && npm publish",
-    "devtool": "tsx ./scripts/devtool.ts"
+    "devtool": "tsx ./scripts/devtool.ts",
+    "check:circulars": "madge --circular --extensions ts --exclude ../../shared ./src"
```

Comment:
[`madge`](https://github.com/pahen/madge?tab=readme-ov-file#circular) is a neat tool which helps us to find circular dependencies. Typescript will actually try to resolve circular dependencies, but they are bad in public packages as they can cause the TS-Compiler to choke when building the AST. This impacts DX both internally and externally.

We should probably add `madge` circular dep checks to all our public packages. cc @LetItRock